### ISSUE TITLE
statuscake_uptime_check created but error on terraform apply

### ISSUE THREAD

**Issue by Eagle-One1** [StatusCakeDev/terraform-provider-statuscake#66]
**Describe the bug**
Error while applying creation resource of type statuscake_uptime_check
The resource is successfully created (can be seen in statuscake web interface) but an error is thrown after creation and nothing is added to the terraform.tfstate

```bash
module.statuscake.statuscake_uptime_check.example_com: Creating...
╷
│ Error: Request cancelled
│ 
│   with module.statuscake.statuscake_uptime_check.example_com,
│   on statuscake/statuscake_tests.tf line 1, in resource "statuscake_uptime_check" "example_com":
│    1: resource "statuscake_uptime_check" "example_com" {
│ 
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-statuscake_v2.0.0 plugin:

panic: reflect: call of reflect.Value.Interface on zero Value

goroutine 27 [running]:
reflect.valueInterface({0x0, 0x0, 0x40d247}, 0x8)
        reflect/value.go:1369 +0x10e
reflect.Value.Interface(...)
        reflect/value.go:1364
github.com/StatusCakeDev/terraform-provider-statuscake/internal/provider.stringElem({0xb17f80, 0x0})
        github.com/StatusCakeDev/terraform-provider-statuscake/internal/provider/utils.go:37 +0xbe
github.com/StatusCakeDev/terraform-provider-statuscake/internal/provider.flattenUptimeCheckRequestHeaders({0xb17f80, 0x0}, 0xc42bd8)
        github.com/StatusCakeDev/terraform-provider-statuscake/internal/provider/resource_uptime_check.go:1114 +0x45
github.com/StatusCakeDev/terraform-provider-statuscake/internal/provider.flattenUptimeCheckHTTPCheck({0xc30740, 0xc0001b6e20}, 0x9)
        github.com/StatusCakeDev/terraform-provider-statuscake/internal/provider/resource_uptime_check.go:964 +0x5be
github.com/StatusCakeDev/terraform-provider-statuscake/internal/provider.resourceStatusCakeUptimeCheckRead({0xd4d4a8, 0xc000618b40}, 0xc000402e00, {0xc353a0, 0xc0004a4780})
        github.com/StatusCakeDev/terraform-provider-statuscake/internal/provider/resource_uptime_check.go:505 +0xad5
github.com/StatusCakeDev/terraform-provider-statuscake/internal/provider.resourceStatusCakeUptimeCheckCreate({0xd4d4a8, 0xc000618b40}, 0xc0001b7668, {0xc353a0, 0xc0004a4780})
        github.com/StatusCakeDev/terraform-provider-statuscake/internal/provider/resource_uptime_check.go:471 +0xe66
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc00029d420, {0xd4d438, 0xc0003f6440}, 0x2, {0xc353a0, 0xc0004a4780})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:341 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00029d420, {0xd4d438, 0xc0003f6440}, 0xc000430dd0, 0xc000402c80, {0xc353a0, 0xc0004a4780})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/resource.go:467 +0x871
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0002907c8, {0xd4d438, 0xc0003f6440}, 0xc00042a640)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/grpc_provider.go:977 +0xd8a
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000328be0, {0xd4d4e0, 0xc0003fa1e0}, 0xc000506000)
        github.com/hashicorp/terraform-plugin-go@v0.7.0/tfprotov5/tf5server/server.go:810 +0x5eb
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xc0d560, 0xc000328be0}, {0xd4d4e0, 0xc0003fa1e0}, 0xc000618180, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.7.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002ae700, {0xd5a810, 0xc00040c1a0}, 0xc000012000, 0xc0003d10b0, 0x12565e0, 0x0)
        google.golang.org/grpc@v1.43.0/server.go:1282 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc0002ae700, {0xd5a810, 0xc00040c1a0}, 0xc000012000, 0x0)
        google.golang.org/grpc@v1.43.0/server.go:1616 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.43.0/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.43.0/server.go:919 +0x294

Error: The terraform-provider-statuscake_v2.0.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**To Reproduce**
1. Download this repository https://github.com/Eagle-One1/statuscake
2. Download status cake provider 2.0.0-pre
3. Unzip the content in ~/.terraform.d/plugins/registry.terraform.io/statuscakedev/statuscake/2.0.0/linux_amd64/terraform-provider-statuscake_v2.0.0
4. run terraform init in the previously cloned repository
5. copy terraform.tfvars.sample to terraform.tfvars
6. replace statuscake_api_key in terraform.tfvars
7. run terraform apply and try to apply

**Expected behavior**
statuscake_uptime_check deployed with no errors and added to terraform.tfstate

**Additional context**
Trying to test the provider, tell me if you need more details

**Comment by tomasbasham** [comment]
From our investigation on this the issue is when a test is created without a request payload. Of course this is not expected and should be allowed to be blank. We're working on a fix for this

**Comment by tomasbasham** [comment]
This has been fixed in version 2.0.1-pre. It is still in beta and will hopefully be in GA shortly
